scripts/html_embeddings/chunk_html.py

```diff
@@ -0,0 +1,414 @@
+    """
+    Chunk a single HTML file.
+
+    Args:
+        input_file: Path to input HTML file
+        output_dir: Directory to save chunks
+        input_base_dir: Base directory for input files (for relative path calculation)
+        max_token_limit: Maximum tokens per chunk
+        count_tag_tokens: Whether to count HTML tags
+        keep_siblings_together: Keep sibling sections together
+        prepend_parent_section_text: Prepend parent section text
+
+    Returns:
+        Tuple of (success, chunk_count)
+    """
+    logger = logging.getLogger(__name__)
+
+    try:
+        with open(input_file, "r", encoding="utf-8") as f:
+            html_content = f.read()
+
+        if not html_content.strip():
+            logger.warning(f"Empty file: {input_file}")
+            return True, 0
+
+        chunks = chunk_html(
+            html_content=html_content,
+            max_token_limit=max_token_limit,
+            count_tag_tokens=count_tag_tokens,
+            keep_siblings_together=keep_siblings_together,
+            prepend_parent_section_text=prepend_parent_section_text,
+        )
+
+        if not chunks:
+            logger.warning(f"No chunks generated for {input_file}")
+            return True, 0
+
+        relative_path = input_file.relative_to(input_base_dir)
+        base_metadata = extract_metadata_from_path(relative_path)
+
+        chunk_count = 0
+        for i, chunk_content in enumerate(chunks):
+            chunk_metadata = {
+                **base_metadata,
+                "chunk_index": i,
+                "total_chunks": len(chunks),
+                "token_count": count_html_tokens(chunk_content, count_tag_tokens),
+                "source_file": str(relative_path),
+            }
+
+            chunk_data = {
+                "id": f"{base_metadata['doc_id']}_chunk_{i:04d}",
+                "content": chunk_content,
+                "metadata": chunk_metadata,
+            }
+
+            chunk_filename = f"{base_metadata['doc_id']}_chunk_{i:04d}.json"
+            chunk_file_path = output_dir / chunk_filename
+
+            with open(chunk_file_path, "w", encoding="utf-8") as f:
+                json.dump(chunk_data, f, indent=2, ensure_ascii=False)
+
+            chunk_count += 1
+
+        return True, chunk_count
+
+    except Exception as e:
+        logger.error(f"Error chunking {input_file}: {e}")
+        return False, 0
+
+
+def extract_metadata_from_path(file_path: Path) -> Dict[str, Any]:
+    """
+    Extract metadata from file path.
+
+    Args:
+        file_path: Relative path to the file
+
+    Returns:
+        Dictionary with extracted metadata
+    """
+    parts = file_path.parts
+
+    if len(parts) >= 2 and parts[-1] == "index.html":
+        doc_name = parts[-2]
+    else:
+        doc_name = file_path.stem
+
+    doc_id = doc_name.replace(" ", "_").replace("-", "_").lower()
+
+    version = None
+    for part in parts:
+        if part.startswith("4.") and len(part) <= 5:
```

Comment:
would be nice to put a comment what these constants means. Right now it is openshift related, but later when OS 5.x will be released, it would be nice to be able to spot these things. Or better - but the "4." in constants section (or constants.py) dunno